environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -183,15 +188,15 @@ private EnvironmentDto updateAzureResourceEncryptionParameters(String accountId,
                 CreatedDiskEncryptionSet createdDiskEncryptionSet = environmentEncryptionService.createEncryptionResources(
                         environmentDtoConverter.environmentToDto(environment));
                 azureParameters.setDiskEncryptionSetId(createdDiskEncryptionSet.getDiskEncryptionSetId());
+                azureDiskEncryptionParametersRepository.save(azureParameters);
             } catch (Exception e) {
                 throw new BadRequestException(e);
             }
             LOGGER.debug("Successfully created the Disk encryption set for the environment {}.", environmentName);
         } else {
             throw new BadRequestException(validateKey.getFormattedErrors());
         }
-        Environment saved = environmentService.save(environment);
-        return environmentDtoConverter.environmentToDto(saved);
+        return environmentDtoConverter.environmentToDto(environment);
```

Comment:
Why this change?

Reply:
❗  why is the environment save removed?

Reply:
AzureParametersRepository.save(azureParameters) saves the parameters and cascades the changes to environment. So, I removed this extra save of environment.